pkg/clients/clients.go

```diff
@@ -162,102 +165,45 @@ func (dcs *DynamicClientSet) gvkForKind(kind Kind) (gvk schema.GroupVersionKind,
-		MutatingWebhookConfiguration, Namespace, PersistentVolume, PodSecurityPolicy, PriorityClass,
-		StorageClass, ValidatingWebhookConfiguration:
-		return false, nil
-	case ControllerRevision, ConfigMap, CronJob, DaemonSet, Deployment, Endpoints, HorizontalPodAutoscaler,
-		Ingress, Job, LimitRange, NetworkPolicy, PersistentVolumeClaim, Pod, PodDisruptionBudget, PodTemplate,
-		ReplicaSet, ReplicationController, ResourceQuota, Role, RoleBinding, Secret, Service, ServiceAccount,
-		StatefulSet:
-		return true, nil
-	}
-
-	resources, err := dcs.getServerResources(gvk.GroupVersion())
-	if err != nil && len(resources) == 0 {
-		// Only return an error here if no resources were returned. Otherwise, process the partial list
-		// and return an error if no match is found.
+	resourceList, err := dcs.getServerResourcesForGV(gvk.GroupVersion())
+	if err != nil {
 		return false, err
 	}
 
-	for _, gvResources := range resources {
-		if gvResources.GroupVersion == gvk.GroupVersion().String() {
-			for _, resource := range gvResources.APIResources {
-				if resource.Kind == gvk.Kind {
-					return resource.Namespaced, nil
-				}
-			}
+	for _, resource := range resourceList.APIResources {
+		if resource.Kind == gvk.Kind {
+			return resource.Namespaced, nil
 		}
 	}
 
 	return true, fmt.Errorf("failed to discover namespace info for %s", gvk)
 }
 
-func (dcs *DynamicClientSet) getServerResources(gvs ...schema.GroupVersion,
-) (resourceLists []*v1.APIResourceList, err error) {
-
-	if len(gvs) > 0 {
-		var resourceList *v1.APIResourceList
-		for _, gv := range gvs {
-			resourceList, err = dcs.getServerResourcesForGV(gv)
-			if err != nil {
-				if discovery.IsGroupDiscoveryFailedError(err) {
-					// Ignore Group Discovery Failed errors. This type of error may cause
-					// later resource introspection to fail, but the error will be handled at that point.
-					err = nil
-				} else {
-					return nil, err
-				}
-			}
-
-			resourceLists = append(resourceLists, resourceList)
-		}
-	} else {
-		resourceLists, err = dcs.DiscoveryClientCached.ServerPreferredResources()
-		if err != nil {
-			if discovery.IsGroupDiscoveryFailedError(err) {
-				// Ignore Group Discovery Failed errors. This type of error may cause
-				// later resource introspection to fail, but the error will be handled at that point.
-				err = nil
-			} else {
-				return nil, err
-			}
-		}
-	}
-
-	return
-}
-
 func (dcs *DynamicClientSet) getServerResourcesForGV(gv schema.GroupVersion,
-) (resourceList *v1.APIResourceList, err error) {
-	resourceList, err = dcs.DiscoveryClientCached.ServerResourcesForGroupVersion(gv.String())
+) (*v1.APIResourceList, error) {
+	resourceList, err := dcs.DiscoveryClientCached.ServerResourcesForGroupVersion(gv.String())
 
 	if err != nil && isServerCacheError(err) {
 		dcs.RESTMapper.Reset()
 
 		resourceList, err = dcs.DiscoveryClientCached.ServerResourcesForGroupVersion(gv.String())
+		if err != nil {
+			return nil, err
+		}
 	}
```

Comment:
I'm confused about this conditional. Why can we ignore any error that isn't `isServerCacheError`?

Reply:
Fixed in subsequent commit